config/asgi.py

```diff
@@ -17,6 +18,29 @@
 # TODO: from channels.security.websocket import AllowedHostsOriginValidator
 from django.core.asgi import get_asgi_application
 
+# Here we setup a debugger if this is desired. This obviously should not be run in production.
+debug_mode = os.environ.get("DEBUG")
+if debug_mode:
+    logging.info(f"Django is set to use a debugger. Provided debug mode: {debug_mode}")
+    if debug_mode == "pycharm-pydevd":
+        logging.info("Entering debug mode for pycharm, make sure the debug server is running in PyCharm!")
+
+        import pydevd_pycharm
+
+        pydevd_pycharm.settrace("host.docker.internal", port=3003, stdoutToServer=True, stderrToServer=True)
+
+        logging.info("Debugger started.")
+    elif debug_mode == "debugpy":
+        logging.info("Entering debug mode for debugpy (VSCode)")
+
+        import debugpy
+
+        debugpy.listen(("0.0.0.0", 3000))
+
+        logging.info("Debugger listening on port 3000.")
+    else:
+        raise ValueError(f"Invalid debug mode given: {debug_mode}")
```

Comment:
I don't think this should be a fatal error, given that the env var is for the whole stack (i.e. maybe another container needs a DEBUG variable, or it's already defined in the environment).

I think you should either call it something more scoped (e.g. `DJANGO_DEBUG`), or just log an error and move on.

Reply:
Good point. I thought that having the applications fail hard would be nice to force people to look at logs (and have it be the last log, since our logging is a bit noisy locally) but this is a good point. 

Addressed in the latest commit.